app/src/utils/utils.ts

```diff
@@ -129,6 +129,24 @@ export function getCurrentTokenClaim(
     : defaultValue;
 }
 
+/**
+ * @function getCurrentTokenUsername
+ * Parses currentUser object's identity provider to return their username
+ * @param {object} currentUser The express request currentUser object
+ * @returns {string | undefined} The username in currentUser or undefined
+ */
+export function getCurrentTokenUsername(currentUser: CurrentUser | undefined): string {
+  if (currentUser?.tokenPayload) {
+    const idpList = readIdpList();
+    const payload = currentUser.tokenPayload as JwtPayload;
+
+    const usernameKey = idpList.find((x) => x.idp === payload.identity_provider)?.username;
+
+    if (usernameKey && usernameKey in payload) return payload[usernameKey];
+  }
+  return 'UnknownUser';
```

Comment:
Should we return `undefined` instead? Easier to check if the parse failed and throw an error.